contrib/shibuqam/examples/shibuqamoauth.nit

```diff
@@ -0,0 +1,157 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Server that implements an authentication bound to a `shibuqam` valid server.
+#
+# # Use as a web service
+#
+# ## Request
+#
+# Two GET fields
+#
+# * `redir`: A full URI that will be used to redirect the user once the authentication is done.
+# * `state`: A temporary string used to check that the callback is legitimate.
+#
+# On request, the service authenticate the user then redirect to the callback with the user informations.
+#
+# ## Response
+#
+# After the authentication, the response is a redirection to `redir` with the following GET arguments:
+#
+# * `id`: the user identifier. Its the code permanent for students. E.g. `"DOEJ00000000"`
+# * `name`: the full name. E.g. `"Doe, John"`
+# * `email`: the UQAM email. E.g. `"doe.john@courrier.uqam.ca"`
+# * `avatar`: the gravatar based on `email`
+# * `state`: the same state to check that the response is not forged.
+#
+# ## Errors
+#
+# If the request is badly formed or if the `redir` is not allowed, then there is no redirection and a text error message is send to the user.
+#
+# If there is a problem during the authentication, then there is a redirection but the GET fields are only `state` and `error`, where `error` is an error message.
+#
+# # To configure and execute
+#
+# ## Run the server
+#
+#    shibuqamoauth authorized_list [host] [port]
+#
+# `authorized_list` is a textfile that registers the list of accepted `redir`.
```

Comment:
I don’t know what you try to block with this, but I don’t think it does anything significant.